azerothcore/azerothcore-wotlk
# fix(Core/Unit): Fix adds in formation not attacking when leader killed

## Changes Proposed:
When a leader in a formation dies it finna reset (initialize) the motion list of the unit that are part of its formation (in this case the two rogues). In this case what happen is that when u whoop Caliph ass into oblivion it gon remove the ChaseMovementGenerator from the rogues and these two bitches won't be able to attack u unless u go back in melee to them. They can't chase yo ass cuz they previously got their chase movement deleted by the leader dying so that's why they be there chillin.

I introduced a new boolean to the ResetFormation method so that I can decide whatever to re-initialize the motion of the adds.

This problem affect erry mob that is in a formation (only affect adds never the leader). Tested also on some other mobs that are in formation like beastmasters and dey hounds in SM.

For reference the table in database that say who is in a formation and who is not is this: creature_formations

## Issues Addressed:

- https://github.com/chromiecraft/chromiecraft/issues/1392
- https://github.com/azerothcore/azerothcore-wotlk/issues/7317

## SOURCE:
Idk what I should put in this mf

## Tests Performed:

- Tested in game with no errors in windows. 


## How to Test the Changes:


1. hop on a mage
2. ` .go creature 23286` 
2. start attacking the trio (caliph + the two rogues)
3. frost nova and move yo ass away from melee while they rooted
4. kill caliph
5. mf rogues should keep attacking instead of chillin in dey place


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
